### ISSUE TITLE
Possible to Install on Bebop 2?

### ISSUE THREAD

**Issue by teja235** [uavpal/disco4g#7]
I’m reading through the code and at first glance, it doesn’t appear that there’s any reason this mod wouldn’t work on Parrots Bebop 2 as well. Do you know if any porting would be required?

**Comment by Telepilote** [comment]
I had wondered the same, although the long distance benefits are less this mod would reduce drop outs as the operator will not need to maintain line of sight and will add tracking for recovery. Both very handy improvements.

**Comment by teja235** [comment]
The WiFi signal from Skycontroller to Bebop is very unreliable and requires direct line of site between controller and drone. LTE maybe a better option than WiFi in a location with a large number of obstacles blocking the signal, or in the case of trying to fly the drone beyond line of sight.

**Comment by teja235** [comment]
Any information about the technical aspects? I can modify the code But this is my first time coding on parrots platform. What’s the difference between disco and bebop firmware?

**Comment by mainframe** [comment]
We dont have Bebop2 to test/verify - but as long its CHUCK and similar firmware it should be possible - and maybe even without any porting.

**Comment by Telepilote** [comment]
It’s not CHUCK so there will be differences and similarities.

**Comment by teja235** [comment]
@mainframe see that’s what I thought. Will try with bebop 2 right after I get it working on my disco - right after finals week. 😓

**Comment by mainframe** [comment]
Bebop 2 hardware is ARM based? Does it have USB port?

**Comment by mainframe** [comment]
This spec seems similar to CHUCK - ie its still P7 SoC and Linux OS on ARM.

**Comment by teja235** [comment]
It is the same SoC, among many other shared components, so I’m positive it will work but maybe not “as is”. I’m waiting on my modem to arrive - gonna get to work figuring this out soon!

**Comment by Telepilote** [comment]
Bebop 2 has a single micro USB port accessible at the rear for updates, image transfer, increase storage capacity and even operate a thermal camera. https://www.parrot.com/business-solutions-us/parrot-professional/parrot-bebop-pro-thermal

**Comment by KillahDT** [comment]
Will this brick my Bebop 2 if i try to install this mod?
Very nice work ice searched for it.
Ive also a Bebop 1 but i think the system is more different to Bebop 2 and C.H.U.C.K. one the Disco.

greetz

**Comment by teja235** [comment]
I don’t think it would be a permanent brick... worst case it doesn’t work and you’d have to reinstall the firmware. But my plan was to go through the code line by line and see if the files that are being changed are in the same spots if that makes sense lol. I wouldn’t recommend just straight up installin it to see what happens only Bc it’s a $300 drone, but if you’re going to do it then please let me know what happens.

**Comment by KillahDT** [comment]
i can test it but how to unbrick if it dont work

**Comment by teja235** [comment]
It kinda depends on what happens lol. You could just do it and if it flops
we’ll figure out how to fix it then 🤷🏽‍♂️

**Comment by uavpal** [comment]
You can try it on Bebop2, I don't see any reason why it would not work. There might be minor changes required to the scripts (apart from replacing text outputs "disco" with "bebop"), but it should certainly not brick your drone! I could imagine the custom kernel modules I cross-compiled will not work on the Bebop2. Might require re-compilation. Worst case: it will just not work, I don't foresee a "bricking" scenario, which would render the drone completely useless.
However, do not try the Disco4G scripts on Bebop1 -OR- on Bebop2 with the original Skycontroller (which is also still sold as a bundle). Both will not work for sure (without bigger changes).

There are a few more people asking for it in the Slack workspace. I do not have a Bebop2 myself, but I believe it would be very useful! Unlike the disco, failover scenarios for a quad are not that critical (as they just hover whenever a connection loss occurs).

Please drop us a note if you get it to work. Happy to add it to the instructions and accept pull requests if code changes are required.

Thanks, SoftIce

**Comment by KillahDT** [comment]
I will install it later i have the skycontroller 2 
But i havent a huawei stick to test 
i will buy one ending this month
if the drone still works after install i will update to this thread

**Comment by uavpal** [comment]
great!! 👍 
(please also update if it does not) 😃

**Comment by KillahDT** [comment]
I need help.
i cannot copy the disco4g-master folder to my bebop
Most of the files are copied but many arent.
i get errors on filezilla and winscp on file
usbserial.ko
usb_wwan.ko
xt_tcpudk.ko
x_tables.ko
and more.

greetz

**Comment by uavpal** [comment]
what error? is there enough disk space?

**Comment by KillahDT** [comment]
Yeah the drone has 7 GB free space.
It shows me that file already exist and cannot overwrite the file.
if i delete the disco 4g folder and ty it again its the same problem.
error 553 but im not sure

**Comment by uavpal** [comment]
can you telnet into your bebop and post the output of `df -h` ? thanks

**Comment by KillahDT** [comment]
BusyBox v1.25.0 (2018-02-14 09:37:35 UTC) built-in shell (ash)
Enter 'help' for a list of built-in commands.

/ # df -h
Filesystem                Size      Used Available Use% Mounted on
ubi1:system              42.2M     38.3M      1.7M  96% /
udev                    144.9M    792.0K    144.2M   1% /dev
shm                     145.0M     36.0K    145.0M   0% /dev/shm
tmp                     145.0M     32.0K    145.0M   0% /tmp
ubi0:factory              4.8M    108.0K      4.4M   2% /factory
ubi2:data                 9.0M      8.9M         0 100% /data
ubi2:update              28.0M     32.0K     26.5M   0% /update
/dev/mmcblk0              7.2G    148.4M      7.0G   2% /data/ftp/internal_000
/ #

**Comment by uavpal** [comment]
Yep, as suspected:
/data and therefore /data/ftp has **0MB** free (of total 9MB)
/data/ftp/internal_000 has 148MB free (of total 7.2GB) - it's a different mountpoint
Seems the Disco has more space available in /data/ftp than Bebop2.
The softmod requires 12MB.

I suggest you use FileZilla to delete /disco4g-master.
You can try to use /internal_000 instead. But this will require changes to the manual instructions as well as installer script!

**Comment by KillahDT** [comment]
Thats to hard for me to do 
i havent the knowledge for it

**Comment by uavpal** [comment]
I haven't the bebop2 for it :(

**Comment by KillahDT** [comment]
Should i send my bebop 2 to you?
😅

**Comment by teja235** [comment]
I gottchu @KillahDT. @UAVPal thanks for staying active on this, I’m sure we can figure it out together. I’ll make those changes to the script and give it a run, modems coming in 2 days.

**Comment by KillahDT** [comment]
Thats cool im happy to hear that.
Waiting for your test 😄

**Comment by ToddTsao** [comment]
1st try run fail. Just share my fail experience.
IP set is the same as Disco.
FTP transfer file to BB2: Done
Enable BB2 telnet mode: press 4 times power button after power on.
1st time install process: BB2 space is lock: Fail
-------------------------------------
Creating ppp directory
Creating symlink for ppp-lte settings
Creating symlink udev rule
Creating symlink for ntpd's config file
Remounting filesystem as read-only
Creating zerotier-one directory
mkdir: can't create directory '/data/lib/zerotier-one': No space left on device
Creating symlink for zerotier-one's local config file
ln: /data/lib/zerotier-one/local.conf: No such file or directory
Removing uavpal softmod installation files
------------------------------------------------------------
Add some command to unlock the space.
   1. cd /etc/colibry/common/ 
   2. mount -o remount,rw /
Install process is done after unlock
Reboot BB2
The 4G dongle LED is no light. No driver. 
How to enable usb port?  what's happen?
I don't know.

**Comment by KillahDT** [comment]
Have you a sim card inside the modem?
I doesnt have a 4g dongle yet but maybe its installed correctly on your drone and only the sim have to sit in the dongle

**Comment by ToddTsao** [comment]
I have install the SIM card. I use the same SIM card for my DISCO-4G & BB2. The Disco-4G is no problem.
The 4G dongle is no power on for BB2.

**Comment by KillahDT** [comment]
Maybe the Bebop 2 only needs a simple software modification to the disco4g master mod.

**Comment by ToddTsao** [comment]
KillahDT: I have find a solution to enable the BB2 USB port. 
FYI
https://community.parrot.com/t5/Bebop-Drone/Bebop-USB-port-what-type-is-that-logically-SOLVED/m-p/133214

**Comment by KillahDT** [comment]
Will anybody update the mod for the bebop2?

**Comment by job2310** [comment]
FYI, result of uname -a

Linux Bebop2-104893 3.4.11+ #1 SMP PREEMPT Wed Feb 14 09:44:38 UTC 2018 armv7l GNU/Linux

**Comment by ToddTsao** [comment]
Currently, we finish below:
Copy Disco-4G to /internal000/
Change file path
Install it to BB2
Enable micro USB port (4G modem power on)

Issue:
BB2 still cannot to drive the 4G modem. It's no connect to Zerotier from BB2.

Need help
Any suggest or command for that.

**Comment by job2310** [comment]
@ToddTsao the mod will not work on the BB2 without several kernel modules being recompiled, specifically the modules used for ppp which drive the 4G connection to the mobile network.

**Comment by KillahDT** [comment]
Is it much of work to do this?

**Comment by job2310** [comment]
I believe it takes a bit of time to do.  The main issue is knowing how to compile the modules for the correct kernel version.  That's outside my skillset.  @uavpal has kindly agreed to do it for us as and when he has time.  It may take several attempts to get the modules compiled and working on the BB2.  I'll post back here if/when we make progress.

**Comment by KillahDT** [comment]
Im happy to hear that and will wait for your update.

greetz

**Comment by uavpal** [comment]
![7939564c-90fd-4302-a6a2-d340c5076464](https://user-images.githubusercontent.com/30078740/40895033-befc1fda-67df-11e8-8d01-155c5edd5226.jpeg)

Teaser...

**Comment by AlaskaJason1090** [comment]
I have waited for this moment for so long. Thank you all who helped build this. I cant wait to try this here in 
Alaska. We have tons great places to fly and this will make it so much better.

**Comment by KillahDT** [comment]
🔥🙂🔥

**Comment by Telepilote** [comment]
Given the great support I would like to further encourage users to hit the donate icon on the code page, I’ve also copied the link here: 

https://www.paypal.com/donate/?token=71em4QrgOUbXVs0Smqd9m6Qy6ZQWWP8jB2Am6bwlydmkhbJmFx0jKKOqgdOWtrMtZFID40&country.x=US&locale.x=US

**Comment by KillahDT** [comment]
Donating link doesnt work for me.

**Comment by uavpal** [comment]
You can try it from here: https://github.com/uavpal/disco4g#is-it-really-free-are-you-crazy
If not, what error do you get?

**Comment by OpenUAS** [comment]
Well if you want more ;) https://www.youtube.com/watch?v=qVogmRistcw

**Comment by job2310** [comment]
FYI, Bebop2 ... the method of killing /usr/bin/usb_mode and switching usb to host doesn't work with the latest firmware as /usr/bin/usb_mode no longer exists.  Looks like Parrot are onto us. :)

**Comment by KillahDT** [comment]
Which firmware is supported?
I will look which i have installed after work.

**Comment by job2310** [comment]
![img_4311](https://user-images.githubusercontent.com/7822449/41094238-830c7750-6a45-11e8-90dd-9967f3d7adc8.PNG)

Live streaming via 4G connection to Bebop2.

I've worked out a process where a Raspberry Pi Zero W can be used to connect to the Bebop2 hotspot and share it's internet connection with the Bebop2.  ZeroTier is installed on the Bebop2 to allow the connection to the SC2, same as the Disco4G mod.

**Comment by job2310** [comment]
For anyone feeling particularly brave, I've put together a bebop2 4G install "package".  The zip can be downloaded from the URL below.  Note the installation for the SC2 is the same as the Disco so please follow those instructions to get your SC2 working.  I've put a few rough install instructions in the "installation.txt" file within the zip.  They should get you started.  Happy to try and help where I can.

The primary differences between the disco4g and bebop4g install processes is that on the bebop2 we've got to kill the /usr/bin/usb_mode process in  order to allow the micro usb to recognise our usb modem.  In addtion, due to the lack of available space on the bebop2, the scripts need to be installed into the /data/ftp/internal_000 folder instead of /data/ftp like on the Disco.

https://drive.google.com/drive/folders/1ecgzVzJHvVKQDevQMu8tItp_3U1C4Rcq?usp=sharing

**Comment by teja235** [comment]
I managed to establish a connection but it's basically unusable due to lag and video distortion even though my network speed tests are turning out great. I'm using a Hi-Link stick, does this version incorporate the fix for H modems?

**Comment by teja235** [comment]
![capture](https://user-images.githubusercontent.com/30220868/41301721-93aead7c-6e1d-11e8-9395-77c08f950c83.JPG)

**Comment by job2310** [comment]
Per the above, I have only tested with 3372-153 modem in Stick Mode.  I do not have a hilink modem.

**Comment by job2310** [comment]
Also what version firmware are you running?

**Comment by teja235** [comment]
4.4.2, latest firmware for bebop and skycontroller.
I read that you only tested with stick, just dropping a note so everyone can see my results too :) 
Have you incorporated the fix for hilink modems into this code?

**Comment by job2310** [comment]
4.7.1 is the latest however this mod doesn't work on that yet.  If you're on 4.4.2 that should be ok.  As I said, my modem converted to stick mode.  So no, hilink modems won't work, at the moment.

**Comment by ToddTsao** [comment]
![Image](https://i.imgur.com/XnARNEy.jpg)

I am ready for bebop 2 4G.

**Comment by ToddTsao** [comment]
Thanks again for Job2310.

**Comment by uavpal** [comment]
go have fun and upload some videos! :)

I'll leave the issue open until the code is finalized, fully tested and documentation is ready.

**Comment by job2310** [comment]
@ToddTsao  Glad it seems to be working for you.  Be careful with the position of the stick modem.  Looking at your picture I would worry that with the bebop moving forward the wind might push the stick backwards and actually unplug it from the micro usb port on the Bebop2.  I use 3M dual lock to attach mine to the top of my Bebop2.

![img_4318](https://user-images.githubusercontent.com/7822449/41356574-19f41a44-6f1c-11e8-9ffb-db566f252e0a.jpg)

**Comment by uavpal** [comment]
it works for me. alternatively, you can try the new push notification instead of SMS

**Comment by Telepilote** [comment]
I hope Anafi 4G is possible, will be interesting to see what Parrot did with that one.

https://encrypted-tbn3.gstatic.com/images?q=tbn:ANd9GcRY4f19Y0qSrMqE063BTrn5KABtzXbhcx71VS1ZOCB3UdhK-Jm-2vlAYnDDUw

**Comment by job2310** [comment]
@Telepilote If you buy me an Anafi I'll let you know.  :)

**Comment by Telepilote** [comment]
I recently lost my Bebop 2 due to loss of signal, I tend to fly long distances. Unfortunately this was before the 4G mod was possible. I’ll replace it but not sure if I’ll get another or save for the Anafi. Of course if I do then the 4G mod will certainly help reduce the chance of loosing either model.

**Comment by AlaskaJason1090** [comment]
I just lost my bebop to a tree and when I got my new one from Best Buy stock firmware was 4.4.0, it was forcing me and now it’s the newest 4.4.7... any way to downgrade, or how long do we think it will take to get a 4g mod working for the new firmware. 

Thanks In advance. 

-Jason Murphy

**Comment by uavpal** [comment]
downgrade your BB2: https://github.com/uavpal/beboptwo4g/wiki/Manual-firmware-upgrade-and-downgrade-of-Parrot-Bebop-2

the 4G softmod release for BB2 is around the corner, stay tuned.

**Comment by uavpal** [comment]
we are in open beta: https://uavpal.com/bebop2
anyone keen to try, go ahead :) thanks for your patience!
I'll close this issue.

Please open a new issue in the BebopTwo4G repository or join us on Slack if there's a problem.

SoftIce

**Comment by ymcr** [comment]
I tested with the Bebop - works perfect !
My Anafi is arriving today, I'm not giving it away, but if any expert wishes to give it a try, I'm happy to assist and beta test.
